### DISCUSSION TITLE
Share links with edit permission

### DISCUSSION THREAD

**Discussion by Frando** [outline/outline#3897] [Ideas]
Hi,
it would be nice to have a way to create share links that have editing permission. We use Outline as an internal wiki, where we also write meeting notes and other docs collaboratively. Sometimes, a meeting includes people who do not have an account on our system. For these situations, when writing a doc collaboratively in a meeting, it would be great to have the ability to create an additional share link with editing permissions. (Like "invite guests" or "create link for guest editors" or so). The guest editors could join the editing session immediately and optional set their display name.
Does this sound useful for others too? And would Outline maintainers integrate this if we'd write a PR?
Thanks,
Frando

**Comment by tommoor** [comment]
Tentatively interested! This does sound quite difficult to do given the current assumptions of the system – for example publicly shared views are completely decoupled from collaborative editing right now, there are also many places that require/assume a user account as part of editing presence/edit recording.

If you're able to formulate a plan for how to tackle this I'd be happy to review and provide guidance

**Comment by almereyda** [comment]
This is indeed a requirement that we often find with peers in our network of NGOs, where collaboration happens ad-hoc and often in an unstructured way.

Am similarily happy to see this initiative coming forward, and would like to contribute with ideas, review and testing.

**Comment by ReessKennedy** [comment]
+1

**Comment by dsmrs** [comment]
+1 (duplicate #5012 )

**Comment by matthijskooijman** [comment]
I am also interested in this, but I can see the amount of work required indeed.

> Tentatively interested! This does sound quite difficult to do given the current assumptions of the system – for example publicly shared views are completely decoupled from collaborative editing right now, there are also many places that require/assume a user account as part of editing presence/edit recording.

One (maybe) easy way to handle this could be to have a special "anonymous" user to use for these public sharing connections? That does have a little smell, so depending on the architecture it might not be a great idea.

Another thought would be that it is actually useful to track multiple anonymous users editing at the same time. Taking Etherpad as an example, there all users are anonymous, but can enter a name into a simple text box and then that name is used to track your contributions (and shown in a list of online users). This name is stored in a session or cookie or so, so persists across multiple editing sessions (and I think you can edit the name later). I think this might translate well into Outline - whenever someone opens an editable public link, create a new user (marked as anonymous) to track contributions. Also, storing some session token to not create a new user every time.

The obvious downside is that this might end up with a very big list of users, but that might be acceptable (and could be optional - you could also disable anonymous user tracking and map each of them to a single user, either right away or retroactively if you want to track users while editing but save space later)?

I initially looked for a feature request about this in the issue list and (because it does not exist yet) then tried creating one, which guided me to the discussions, and then I found this. Would it make sense to create a feature request issue about this, even if it is not immediately planned? I'm happy to make it, but given the feature request button explicitly points here, did not want to do without asking here first :-)

**Reply by matthijskooijman** [reply]
This also relates to https://github.com/outline/outline/issues/7362, which probably requires some of the same changes to anonymous user tracking.

**Reply by almereyda** [reply]
Discourse is doing something similar for you when interacting with Email identities via the system. They call it staged users, which can be converted into real users, once their sign up actively with their email identity.

**Reply by m8426** [reply]
Hi @tommoor do you have any thoughts on the idea above? Do you expect any progress on this in the near future?

**Reply by tommoor** [reply]
I'd imagine we'd create an AnonymousUser record separate from real users to avoid any possibility of conflation and require a name as a minimum. Anonymous users would be able to authenticate with the collaboration server if the share settings allow it and therefore edit the doc.

The next ask will be commenting – which also explicitly does not exist on publicly shared documents though - comments are private to the workspace.

I don't think I'll have time to take this on in the near future. The workaround today is simply to invite somebody directly as a guest to the doc which is really not that different.

**Reply by m8426** [reply]
I think the difference is being able to send out a link for anyone to use vs manually inviting users...its a different workflow entirely. I understand its not a priority.

**Reply by m8426** [reply]
@tommoor how strongly do you feel about keeping the guest user feature cloud/paid only? 

I have a feeling that it wouldn't be too difficult to build public editing off of that feature, and I'd be happy to work on that and submit it

**Reply by tommoor** [reply]
I feel strongly this feature will not come to community ed. But also, as I said I would not build this ontop of the existing user record anyway, it's too risky

**Reply by m8426** [reply]
If you change your mind, I would love to contribute towards this feature with the condition that it is available on all plans of course. But if you want to keep it as a paid feature, that's fair, and its up to you.

**Comment by danielraab** [comment]
+1
I saw this relatively old branch https://github.com/outline/outline/tree/share-in-edit-mode.
Is there any plan to implement this functionality ?

**Comment by e1ke** [comment]
+1 I would like to use outline for a wedding as planning collaboration, sharing it via web and allow editing instead of viewing only would be mandatory for this use case. I have a few more use cases where I want to allow editing for anonymous users, so I would highly appreciate this feature
